extensions/spring-xd-extension-http/src/main/java/org/springframework/integration/x/http/NettyHttpInboundChannelAdapter.java

```diff
@@ -286,6 +335,58 @@ public void writeRequested(ChannelHandlerContext ctx, MessageEvent e) throws Exc
 		}
 	}
 
+	/* start whu */
+	void wrapSendMessage(Message<?> message) {
+		sendMessage(message);
+	}
+
+	private boolean sendMessageToResponsibleSender(HttpRequest request, Message<?> message) {
+		String name = "wrapSendMessage";
+		Object sender = getResponsibleSender(request);
+		for (Method m : sender.getClass().getDeclaredMethods()) {
+			if (m.getName().equals(name)) {
+				m.setAccessible(true);
+				try {
+					m.invoke(sender, message);
+					return true;
+				}
+				catch (Exception e) {
+					logger.warn("Exception when sending message", e);
+				}
+			}
+		}
+		return false;
+	}
+
+	@SuppressWarnings("unchecked")
+	private <T> T getResponsibleSender(HttpRequest request) {
+		T result = null;
+		List<String> matches = new LinkedList<>();
+		String uri = request.getUri();
+		Map<String, ?> map = getModuleState();
+		for (String path : map.keySet()) {
+			if (uri.equals(path) || uri.matches(path)) {
+				matches.add(path);
+				result = (T) map.get(path);
+			}
+		}
+		if (matches.size() > 1) {
+			logger.warn("Ambiguous HTTP path, multiple matches " + matches + " for path '" + uri + "'");
+		}
+		return result;
+	}
+
+	@SuppressWarnings("unchecked")
+	private static Map<String, Object> getModuleState() {
+		/* bit hacky, store module state in system properties */
```

Comment:
Thanks for the helpful comments. How can we proceed with this? Did you have a chance to discuss with the team? Some sort of built-in support between shared state between the modules would be ideal.